src/metatrain/experimental/nanopet/trainer.py

```diff
@@ -306,6 +307,22 @@ def train(
                 )
                 targets = average_by_num_atoms(targets, systems, per_structure_targets)
 
+                # as the model may predict empty blocks - exemplar case is for spherical
+                # targets on an atomic basis for blocks corresponding to atomic types
+                # not in the current batch - drop these empty blocks according to the
+                # target keys.
+                for target_name in targets.keys():
+                    if predictions[target_name].keys != targets[target_name].keys:
+                        # TODO: use `metatensor.filter_blocks` once PR #XXX is available
```

Comment:
Is this https://github.com/metatensor/metatensor/pull/885? I can make a release =)

Reply:
Yes! Thanks :)